specs/packagemanager/PackageManagement.md

```diff
@@ -0,0 +1,1134 @@
+        static PackageVolume FindPackageVolumeByMediaId(String mediaId);
+
+        /// Get the specified volume.
+        /// @see https://learn.microsoft.com/en-us/uwp/api/windows.management.deployment.packagemanager.findpackagevolume
+        static PackageVolume FindPackageVolumeByName(String name);
+    };
+
+    /// The progress status of the deployment request.
+    /// @see https://learn.microsoft.com/en-us/uwp/api/windows.management.deployment.deploymentprogress.state?view=winrt-22621
+    [contract(PackageDeploymentContract, 1)]
+    enum PackageDeploymentProgressStatus
+    {
+        Queued = 0,             // The request is queued
+        InProgress = 1,         // The request is in progress
+        CompletedSuccess = 2,   // The request completed successfully
+        CompletedFailure = 3,   // The request failed with some critical internal error.
+    };
+
+    /// Contains progress information for the deployment request.
+    /// @see https://learn.microsoft.com/en-us/uwp/api/windows.management.deployment.deploymentprogress?view=winrt-22621
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageDeploymentProgress
+    {
+        PackageDeploymentProgressStatus status;
+
+        /// The progress percentage of the deployment request.
+        /// @note This is a double with values 0.0-1.0. Windows.Management.Deployment.DeploymentProgress.percentage is uint32 with values 0-100.
+        Double percentage;
+    }
+
+    /// The status of the deployment request.
+    /// @see PackageDeploymentResult.Status
+    [contract(PackageDeploymentContract, 1)]
+    enum PackageDeploymentStatus
+    {
+        InProgress = 0,         // The request is in progress
+        CompletedSuccess = 1,   // The request completed successfully
+        CompletedFailure = 2,   // The request failed with some critical internal error.
+    };
+
+    // Provides the result of a deployment request.
+    // @see https://learn.microsoft.com/en-us/uwp/api/windows.management.deployment.deploymentresult
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageDeploymentResult
+    {
+        PackageDeploymentStatus Status { get; };
+        HRESULT ExtendedError { get; };
+        Boolean IsRegistered{ get; };
+        Guid ActivityId { get; };
+    }
+
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageSetItem
+    {
+        PackageSetItem();
+
+        String Id;
+        String PackageFamilyName;
+        Windows.ApplicationModel.PackageVersion MinVersion;
+        Microsoft.Windows.ApplicationModel.DynamicDependency.PackageDependencyProcessorArchitectures ProcessorArchitectureFilter;
+        Windows.Foundation.Uri PackageUri;
+    }
+
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageSet
+    {
+        PackageSet();
+
+        String Id;
+        IVector<PackageSetItem> PackageSetItems { get; };
+    }
```

Comment:
Needs additional property

`Uri PackageUri;`

PackageSet is an array of PackageSetItems where each item has PackageFamilyName+MinVersion+PackageUri (plus other related but those are the key items). That's awesome when you have multiple msix e.g.

PackageSet has PackageSetItems[3] = {
    { family-foo, 1.2.3.4, file://c:/foo/blah.msix },
    { family-bar, 2.4.6.8, file://d:/stuff/bar.msix },
    { family-meh, 3.3.3.3, https://whatever.com/meh.msix },
}

But in the case of ms-uup: URIs you have 1 URI corresponding to N packages. Per the current IDL you can do this with redundancy

PackageSet has PackageSetItems[3] = {
    { family-foo, 1.2.3.4, ms-uup://Product/one.product.has.multiple.packages.x64 },
    { family-bar, 2.4.6.8, ms-uup://Product/one.product.has.multiple.packages.x64 },
    { family-meh, 3.3.3.3, ms-uup://Product/one.product.has.multiple.packages.x64 },
}

Adding a Uri property to the PackageSet object we can remove the redundancy e.g. `

PackageSet.PackageSetItems[3] = {
    { family-foo, 1.2.3.4  },
    { family-bar, 2.4.6.8 },
    { family-meh, 3.3.3.3 },
}
and PackageSet.Uri=ms-uup://Product/one.product.has.multiple.packages.x64.

Treat PackageSet.PackageSetItem[n].Uri is an _override_ for PackageSet.Uri -- if PackageDeploymentManager needs a URI it uses PackageSetItem.Uri if defined, else PackageSet.Uri (else fail)

(Came up after spec posted for review else the spec would already have this) 
RECOMMEND: Add `Uri PackageUri;` to PackageSet